File: sdks/dotnet/tests/Mobiscroll.Connect.Tests/TestHelpers/ClientFactory.cs
using System.Net.Http;
using Mobiscroll.Connect;

namespace Mobiscroll.Connect.Tests.TestHelpers;

internal static class ClientFactory
{
    public static MobiscrollConnectClient Create(FakeHttpMessageHandler handler, string baseUrl = "https://connect.mobiscroll.com/api/")
    {
        var http = new HttpClient(handler);
        var config = new MobiscrollConnectConfig
        {
            ClientId = "test-client-id",
            ClientSecret = "test-client-secret",
            RedirectUri = "http://localhost:5000/callback",
        };
        return new MobiscrollConnectClient(config, http, baseUrl);
    }
}

File: sdks/java/connect-sdk/src/test/java/com/mobiscroll/connect/support/ClientFactory.java
package com.mobiscroll.connect.support;

import com.mobiscroll.connect.MobiscrollConnectClient;
import com.mobiscroll.connect.MobiscrollConnectConfig;
import com.mobiscroll.connect.models.TokenResponse;
import okhttp3.mockwebserver.MockWebServer;

/** Helpers to build a {@link MobiscrollConnectClient} pointed at a {@link MockWebServer}. */
public final class ClientFactory {

    private ClientFactory() {}

    public static MobiscrollConnectClient withMock(MockWebServer server) {
        return new MobiscrollConnectClient(MobiscrollConnectConfig.builder()
                .clientId("test-client")
                .clientSecret("test-secret")
                .redirectUri("http://localhost/callback")
                .baseUrl(server.url("/api").toString())
                .build());
    }

    public static MobiscrollConnectClient withMockAndCredentials(MockWebServer server) {
        MobiscrollConnectClient c = withMock(server);
        c.setCredentials(new TokenResponse("access-1", "Bearer", 3600, "refresh-1", "calendars events"));
        return c;
    }
}
